### ISSUE TITLE
Add i18n support

### ISSUE THREAD

**Issue by gaecoli** [RedashCommunity/redash#62]
We use GitHub only for bug reports 🐛

Anything else should be a discussion: https://github.com/getredash/redash/discussions/ 👫

🚨For support, help & questions use https://github.com/getredash/redash/discussions/categories/q-a
💡For feature requests & ideas use https://github.com/getredash/redash/discussions/categories/ideas

Alternatively, check out these resources below. Thanks! 😁.

- [Discussions](https://github.com/getredash/redash/discussions/)
- [Knowledge Base](https://redash.io/help)

**Comment by justinclift** [comment]
Yeah, that's a good idea.  Most of the multilingual stuff I've seen before uses some form of token and look up table to achieve it.

That doesn't always work out well for languages that are right-to-left (eg Arabic, Hebrew), or those which are vertical (Chinese, Korean, Japanese).  Doing a quick search online now, it seems like at least Chinese, Korean, and Japanese are ok to be written horizontally these days, so that might not be all that much of a problem.

For our needs, we probably only need to do the web interface stuff.  So, a javascript only based system (rather than needing to include Python) should be fine.

Thoughts? :smile:

**Comment by gaecoli** [comment]
> 

Yes. Just javascript.

**Comment by justinclift** [comment]
Hmmm, to me that looks like the wrong end of it.

Shouldn't we use [Launchpad](https://translations.launchpad.net) or an existing online service for the actual translation?

I was more thinking about using a library for doing the text lookup and string display in our web interface.

**Comment by gaecoli** [comment]
> Hmmm, to me that looks like the wrong end of it.
> 
> Shouldn't we use [Launchpad](https://translations.launchpad.net) or an existing online service for the actual translation?
> 
> I was more thinking about using a library for doing the text lookup and string display in our web interface.

You are right!

**Comment by justinclift** [comment]
Cool.  As a data point, my mention of Launchpad above is only because I used to use it for translation strings many years ago.

I haven't touched it since then though, so it's completely possible there are much better options around. :smile:
